sources/api/netdog/src/interface_name.rs

```diff
@@ -0,0 +1,195 @@
+//! The interface_name module contains the definition of a valid network interface name and the
+//! deserializtion code to support creation of the structure from string.
+//!
+//! A valid network interface name is defined by the criteria in the linux hernel:
+//! https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/net/core/dev.c#n988
+use crate::net_config::NetInterface;
+use indexmap::IndexMap;
+use serde::{
+    de::{Error as DeError, MapAccess, Visitor},
+    Deserializer, Serialize, Serializer,
+};
+use snafu::ensure;
+use std::{convert::TryFrom, fmt, ops::Deref};
+use toml::Value;
+
+/// InterfaceName can only be created from a string that contains a valid network interface name.
+/// Validation is handled in the `TryFrom` implementation below.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub(crate) struct InterfaceName {
```

Comment:
It looks like you can get rid of the entire `deserialize_interfaces_map` function by using the `TryFrom<&str>` impl for deserialization.

```suggestion
#[derive(Debug, Clone, Eq, PartialEq, Hash, Deserialize)]
#[serde(try_from = "&str")]
pub(crate) struct InterfaceName {
```

I tried this with a local checkout of your branch and all the tests passed, after removing the function and the field deserializer in `NetConfig`.

Reply:
You're right!  Thanks!